lib/request_overrider.js

```diff
@@ -89,18 +89,16 @@ function RequestOverrider(req, options, interceptors, remove, cb) {
     )
   }
 
-  if (!req.connection) {
-    req.connection = new EventEmitter()
-  }
-
   req.path = options.path
   req.method = options.method
 
   options.getHeader = function(name) {
     return getHeader(req, name)
   }
 
```

Comment:
Could you add a comment here with some of the info from the top comment in your PR, and also a test that checks that these aliases are `Object.is` the same thing?